src/Compilers/CSharp/Test/Symbol/Symbols/LocalFunctionTests.cs

```diff
@@ -189,5 +189,116 @@ public void M()
                 // }
                 Diagnostic(ErrorCode.ERR_EOFExpected, "}").WithLocation(7, 1));
         }
+
+        [Fact, WorkItem(65938, "https://github.com/dotnet/roslyn/issues/65938")]
+        public void StaticLocalFunction_CapturingMethodGroup()
+        {
+            CreateCompilation("""
+                var c = new C();
+                LocalFunc();
+
+                static void LocalFunc()
+                {
+                    var x = c.MyExtension;
+                    x();
+                }
+
+                public class C
+                {
+                }
+
+                public static class Extensions
+                {
+                    public static string MyExtension(this C c)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics(
+                    // (6,13): error CS8421: A static local function cannot contain a reference to 'c'.                       
+                    //     var x = c.GetName;
+                    Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureVariable, "c").WithArguments("c").WithLocation(6, 13));
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup2()
+        {
+            CreateCompilation("""
+                var c = new C();
+                LocalFunc();
+
+                static void LocalFunc()
+                {
+                    var x = Extensions.MyExtension;
+                    x(c);
+                }
+
+                public class C
+                {
+                }
+
+                public static class Extensions
+                {
+                    public static string MyExtension(this C c)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics(
+                    // (7,7): error CS8421: A static local function cannot contain a reference to 'c'.
+                    //     x(c);
```

Comment:
>//     x(c); [](http://example.com/codeflow?start=20&length=12)

Let's remove this code, which is not about method group conversions, change this scenario into success case, and verify that we can compile it successfully.